polynote-kernel/src/main/scala/polynote/kernel/logging/package.scala

```diff
@@ -0,0 +1,142 @@
+package polynote.kernel
+
+import java.io.PrintStream
+
+import polynote.env.ops.Location
+import zio.{Has, UIO, URIO, ZIO, ZLayer}
+import zio.blocking.Blocking
+
+import scala.collection.immutable.StringOps
+
+package object logging {
+  type Logging = Has[Logging.Service]
+
+  object Logging {
+
+    def live: ZLayer[Blocking, Nothing, Logging] = ZLayer.fromService {
+      blocking: Blocking.Service => new Logging.Service.Default(System.err, blocking)
+    }
+
+    trait Service {
+      def error(msg: String)(implicit location: Location): UIO[Unit]
+      def error(msg: Option[String], err: Throwable)(implicit location: Location): UIO[Unit]
+      def error(msg: Option[String], err: zio.Cause[Throwable])(implicit location: Location): UIO[Unit]
+      def warn(msg: String)(implicit location: Location): UIO[Unit]
+      def info(msg: String)(implicit location: Location): UIO[Unit]
+      def remote(msg: String): UIO[Unit]
+    }
+
+    object Service {
+
+      class Default(out: PrintStream, blocking: Blocking.Service) extends Service {
+        private val Red = "\u001b[31m"
+        private val Reset = "\u001b[0m"
+        private val remotePrefix = "[REMOTE] "
```

Comment:
Not related to this PR but just because I'm looking at this code now:

Maybe we should try printing which kernel the remote log is coming from? Sometimes when debugging it's hard to tell what's generating a particular log if there are multiple kernels up.